File: src/utils/validator.ts
export function validateAlgorithm(jwt: string): boolean {
    const parts = jwt.split('.');
    if (parts.length !== 3) {
        throw new Error('Invalid JWT format');
    }

    const header = JSON.parse(Buffer.from(parts[0], 'base64url').toString());
    return header.alg !== 'none';
}

File: src/utils/validator.py
import jwt
from typing import Dict, Any

def has_none_algorithm(decoded_jwt: Dict[str, Any]) -> bool:
    #Check if JWT uses 'none' algorithm.
    return decoded_jwt.get("header", {}).get("alg") == "none"

def validate_jwt_structure(token: str) -> bool:
    #Validate basic JWT structure.
    parts = token.split('.')
    return len(parts) == 3